nav2_util/src/odometry_utils.cpp

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) 2018 Intel Corporation
+// Copyright (c) 2020 Sarthak Mittal
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <string>
+
+#include "nav2_util/odometry_utils.hpp"
+
+using namespace std::chrono;  // NOLINT
+using namespace std::chrono_literals;  // NOLINT
+
+namespace nav2_util
+{
+
+OdomSmoother::OdomSmoother(
+  rclcpp::Node::SharedPtr nh,
+  double filter_duration)
+: node_(nh),
+  odom_history_duration_(rclcpp::Duration::from_seconds(filter_duration))
+{
+  std::string odom_topic;
+  nh->get_parameter_or("odom_topic", odom_topic, std::string("odom"));
```

Comment:
This parameter is never declared, so it will always fail to change the topic. Because this is a utility, my recommendation is making this a constructor parameter for the user of this object to provide and set the topic as a parameter in the bt navigator and declare it (and add to yaml).  Then you can get in the speed controller and pass to this utility.